lib/streamaggr/streamaggr.go

```diff
@@ -875,7 +905,8 @@ func (a *aggregator) Push(tss []prompbmarshal.TimeSeries, matchIdxs []byte) {
 		}
 	}
 	if totalSamples > 0 {
-		a.samplesLag.Update(float64(totalLag/int64(totalSamples)) / 1000)
+		a.matchedSamples.Add(totalSamples)
+		a.samplesLag.Update(float64(totalLag/int64(totalSamples)) / 1_000)
```

Comment:
@AndrewChubatiuk wdyt about measuring not avg (which I don't know how to use) but max lag (which would show how close we're to the limit)?
I'm not rly sure with this suggestion, so I'd ask what what data would you prefer to have before making decision about increasing aggr interval: the max or avg lag? Or do you think this metric is valuable at all? What kind of decision can you make based on it?

Reply:
updated to max lag